### PR TITLE
MM-17149 - Extend config.json for marketplace settings

### PR DESCRIPTION
**Pull request by ali-farooq0**
#### Summary
Extend config.json to include plugin marketplace settings.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-17149

#### Related changes
Webapp changes https://github.com/mattermost/mattermost-webapp/pull/3503

### CODE REVIEW

File: model/config.go [hunk 1]
```diff
@@ -2201,6 +2203,8 @@ type PluginSettings struct {
 	ClientDirectory          *string `restricted:"true"`
 	Plugins                  map[string]map[string]interface{}
 	PluginStates             map[string]*PluginState
+	EnableMarketplace        *bool
+	MarketplaceURL           *string
```
Comment (ali-farooq0) [inline model/config.go]: went with `URL` as opposed to `Url`, which unfortunately is inconsistent with 
https://github.com/mattermost/mattermost-server/pull/11933/files#diff-2279f86634da3f787f279961af76713aR2200 `AllowInsecureDownloadUrl` 🤦‍♂ , but is more aligned with go's convention. Breaks my 💔  to see the inconsistency.

File: app/diagnostics.go [hunk 2]
```diff
@@ -602,6 +602,7 @@ func (a *App) trackConfig() {
 		"enable_uploads":                *cfg.PluginSettings.EnableUploads,
 		"allow_insecure_download_url":   *cfg.PluginSettings.AllowInsecureDownloadUrl,
 		"enable_health_check":           *cfg.PluginSettings.EnableHealthCheck,
+		"enable_marketplace":            *cfg.PluginSettings.EnableMarketplace,
```
Comment (lieut-data) [inline app/diagnostics.go]: Can we add diagnostics to track when the marketplace is different from the default marketplace?

File: model/config.go (hunk 1 above)
Comment (lieut-data) [reply model/config.go]: 0/5, but should we keep the inconsistency with the rest of this file/struct?

Comment (ali-farooq0) [reply model/config.go]: there is already a mix of `Url` `URL` within this file, however maybe it's worth keeping it the same within this struct. 👍

File: app/diagnostics.go (hunk 2 above)
Comment (ali-farooq0) [reply app/diagnostics.go]: @lieut-data any objections in always logging `MarketplaceUrl` instead?

Comment (lieut-data) [reply app/diagnostics.go]: @ali-farooq0, actually, yes: we don't want to expose the internal details of a customer's configuration (e.g. the name of their internal host). When it's a free-form field, we just report when it's "different" to improve privacy.

**Pull request merged**